### PR TITLE
feat: add spinnaker_pipeline_template_v2 resource

### PR DESCRIPTION
**Pull request by martinohmann**
Notable changes:

- add `ResponseError` to make working with API errors easier
- add validation to the `template` field of the new
  `spinnaker_pipeline_template_v2` resource to catch common issues on
  `terraform plan`
- tests for error impl and parsing/validation logic

More tests require general changes to the way the API client is
instantiated and passed around. This is out of scope here and will be
addressed in a separate change.

Note that other API calls now could also benefit from `ResponseError`
to simplify the code but changing this is out of scope of this change as well.

### CODE REVIEW

File: spinnaker/api/models.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package api
+
+// PipelineTemplateV2 defines the schema of the pipeline template JSON.
+type PipelineTemplateV2 struct {
+	ID        string                       `json:"id,omitempty"`
+	Metadata  PipelineTemplateV2Metadata   `json:"metadata"`
+	Pipeline  interface{}                  `json:"pipeline"`
```
Comment (pjungermann) [inline spinnaker/api/models.go:7]: What is the reason to not have the schema for Pipeline, too?

File: spinnaker/api/models.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	Scopes      []string `json:"scopes,omitempty"`
+}
+
+type PipelineTemplateV2Variable struct {
+	Name         string      `json:"name"`
+	Description  *string     `json:"description,omitempty"`
+	DefaultValue interface{} `json:"defaultValue,omitempty"`
+	Type         string      `json:"type"`
+	Group        *string     `json:"group,omitempty"`
+	Example      *string     `json:"example,omitempty"`
```
Comment (pjungermann) [inline spinnaker/api/models.go]: I cannot see Group and Example anymore at the V2 spec (https://spinnaker.io/reference/pipeline/templates/) but only at V1 (https://github.com/spinnaker/dcd-spec/blob/master/PIPELINE_TEMPLATES.md#variables). Are they still needed for V2?

File: spinnaker/api/models.go (hunk 1 above)
Comment (martinohmann) [reply spinnaker/api/models.go:7]: I opted to not do that now as it contains lots of fields and subtypes. I'd like to tackle this when refactoring the pipeline resource logic.

For now I just wanted to have the schema for the pipeline template specific top level fields in order to validate them on plan.

However If you think it makes sense to do that now I can spend some time to create the proper types for the pipeline schema.

File: spinnaker/api/models.go (hunk 2 above)
Comment (martinohmann) [reply spinnaker/api/models.go]: You are right. I got confused by the last line on the page and while looking at the V1 docs i thought these fields are also available in V2. I can remove them.

File: spinnaker/api/models.go (hunk 1 above)
Comment (pjungermann) [reply spinnaker/api/models.go:7]: I think it is fine to postpone it a bit as you suggested.

**Review by pjungermann — APPROVED** [review]

**Review by pjungermann — APPROVED** [review]

**Pull request merged**